internal/dao/stubs/pubkey_dao.go

```diff
@@ -19,36 +20,52 @@ func init() {
 	dao.GetPubkeyDao = getPubkeyDao
 }
 
-type ContextReadError struct{}
+type contextReadError struct{}
 
-func (m *ContextReadError) Error() string {
+func (m *contextReadError) Error() string {
 	return "failed to find or convert dao stored in testing context"
 }
 
 func WithPubkeyDao(parent context.Context, init_store []*models.Pubkey) context.Context {
-	ctx := context.WithValue(parent, pubkeyCtxKey, PubkeyDaoStub{init_store})
+	ctx := context.WithValue(parent, pubkeyCtxKey, &PubkeyDaoStub{init_store})
 	return ctx
 }
 
+func PubkeyStubCount(ctx context.Context) (int, error) {
+	pkdao, err := getPubkeyDaoStub(ctx)
+	if err != nil {
+		return 0, err
+	}
+	return len(pkdao.store), nil
+}
+
 func getPubkeyDao(ctx context.Context) (dao.PubkeyDao, error) {
+	return getPubkeyDaoStub(ctx)
+}
+
+func getPubkeyDaoStub(ctx context.Context) (*PubkeyDaoStub, error) {
 	var err error
-	pkdao, ok := ctx.Value(pubkeyCtxKey).(dao.PubkeyDao)
+	pkdao, ok := ctx.Value(pubkeyCtxKey).(*PubkeyDaoStub)
 	if !ok {
```

Comment:
Nitpick: The gold standard in Go is to do ok tests on one line (separated with a semicolon).